clang/test/Modules/pr133057.cpp

```diff
@@ -0,0 +1,143 @@
+// RUN: rm -rf %t
```

Comment:
This test looks scary. Is there any chance at taking some inspiration from `clang/test/SemaCXX/overload-decl.cpp` which is supposed to test the surroundings of this code? 0d1da2298aaa8 also seems to have some relevant tests we could build upon...

Reply:
It is the smallest reproducer we have for the moment. It cannot be integrated into `overload-decl.cpp` because we need modules to trigger deserialization and have overload resolution find different `FunctionDecl`s between constructing the conversion sequence for the copy and the move assignment.

Reply:
I agree with that assessment. Let's move forward then.